docs/test_generate_docs.py

```diff
@@ -56,13 +56,22 @@ def consistency_check(obj, obj_name):
         .union(varkwargs)
     )
 
-    if actual_args.intersection(doc_args) < actual_args:
-        undoc_args = ", ".join(actual_args.difference(doc_args))
+    # Strip `**kwargs` to allow forwarding kwargs to underlying implementations,
+    # or deprecating existing functionality
+    undocumented = actual_args.difference(doc_args).difference({"**kwargs"})
+    # If the sig contains **kwargs, any keyword is valid
+    if any(k.startswith("**") for k in actual_args):
```

Comment:
This logic changed two things:

- If `**kwargs` is an explicit parameter you don't need to document it. This was the main goal here, to avoid documenting the deprecated `**kwargs` in `DaskExecutor`. I've reverted this and just documented it as `DEPRECATED`.
- If `**kwargs` is an explicit parameter, documented parameters that aren't explicit parameters are valid. this wasn't affected in this PR, but was in other PRs I've filed. In a subclass it can be nice to only add keywords with defaults for the new kwargs, and use `**kwargs` to forward the remaining parameters to the base class. With the previous check, all keyword arguments had to be duplicated in the subclasses's init, otherwise documenting them would error with `"Foo has documentation for arguments that aren't real"`. I've left this change.